lib/rpmsg/rpmsg.c

```diff
@@ -219,8 +219,7 @@ struct rpmsg_endpoint *rpmsg_get_endpoint(struct rpmsg_device *rdev,
 			return ept;
 		/* else use name service and destination address */
 		if (name)
-			name_match = !strncmp(ept->name, name,
-					      sizeof(ept->name));
+			name_match = !strcmp(ept->name, name);
```

Comment:
What about using 
`name_match = !strncmp(ept->name, name, RPMSG_NAME_SIZE)`?
We cannot guaranty that the names will finish with the `'\0'` ...

Reply:
sizeof(ept->name) is the same as RPMSG_NAME_SIZE. There's no problem.